velox/expression/SimpleFunctionAdapter.h

```diff
@@ -294,7 +315,27 @@ class SimpleFunctionAdapter : public VectorFunction {
           bits::setNull(nullBuffer, row);
         }
       };
-      if (allNotNull) {
+      if (callNullFree) {
+        // This results in some code duplication, but applying this check once
+        // per batch instead of once per row shows a significant performance
+        // improvement when there are no nulls.
+        if (applyContext.mayHaveNullsRecursive) {
+          applyContext.applyToSelectedNoThrow([&](auto row) INLINE_LAMBDA {
+            typename return_type_traits::NativeType out;
+            bool notNull = !(readers.containsNulls(row) || ...) &&
```

Comment:
callNullFree allows users to return null, i wonder if it would ever do that?
shall we make it by non-null generating.